src/pathx.c

```diff
@@ -2891,6 +2912,25 @@ int pathx_expand_tree(struct pathx *path, struct tree **tree) {
     return -1;
 }
 
+/* Generate a numeric string to use for step->name
+ * Scan tree->children for the highest numbered label, and add 1 to that
+ */
+static char *step_seq_choose_name(struct tree *tree) {
+    int max_node_n=0;
+    char *step_name;
+    for(tree=tree->children; tree!=NULL; tree=tree->next) {
+        if ( tree->label == NULL)
+          continue;
+        int node_n=atoi(tree->label);
```

Comment:
AFAIK, `atoi` should not be used anymore, and `strtoul` would be more appropriate. If you do that, check for overflow

Reply:
I've updated the code to use `strtoul()` instead of `atoi()` and report an overflow if it happens.

Can you please re-approve this Pull Request with the change?

The diff is:
```diff
diff --git a/src/pathx.c b/src/pathx.c
index 4d59d1d4..f2f9c09e 100644
--- a/src/pathx.c
+++ b/src/pathx.c
@@ -35,7 +35,7 @@ static const char *const errcodes[] = {
     "no error",
     "empty name",
     "illegal string literal",
-    "illegal number",
+    "illegal number",                             /* PATHX_ENUMBER */
     "string missing ending ' or \"",
     "expected '='",
     "allocation failed",
@@ -2832,7 +2832,7 @@ static int locpath_search(struct locpath_trace *lpt,
     return result;
 }
 
-static char *step_seq_choose_name(struct tree *tree);
+static char *step_seq_choose_name(struct pathx *path, struct tree *tree);
 
 /* Expand the tree ROOT so that it contains all components of PATH. PATH
  * must have been initialized against ROOT by a call to PATH_FIND_ONE.
@@ -2883,7 +2883,7 @@ int pathx_expand_tree(struct pathx *path, struct tree **tree) {
         if (s->axis != CHILD && s->axis != SEQ)
             goto error;
         if (s->axis==SEQ && s->name == NULL) {
-            s->name = step_seq_choose_name(parent);
+            s->name = step_seq_choose_name(path, parent);
         }
         if (s->name == NULL )
             goto error;
@@ -2914,18 +2914,28 @@ int pathx_expand_tree(struct pathx *path, struct tree **tree) {
 
 /* Generate a numeric string to use for step->name
  * Scan tree->children for the highest numbered label, and add 1 to that
+ * numeric labels may be interspersed with #comment or other labels
  */
-static char *step_seq_choose_name(struct tree *tree) {
-    int max_node_n=0;
+static char *step_seq_choose_name(struct pathx *path, struct tree *tree) {
+    unsigned long int max_node_n=0;
+    unsigned long int node_n;
     char *step_name;
+    char *label_end;
     for(tree=tree->children; tree!=NULL; tree=tree->next) {
         if ( tree->label == NULL)
           continue;
-        int node_n=atoi(tree->label);
+        node_n=strtoul(tree->label, &label_end, 10);
+        if ( label_end == tree->label || *label_end != '\0' )
+          /* label is not a number - ignore it */
+          continue;
+        if ( node_n >= ULONG_MAX ) {
+          STATE_ERROR(path->state, PATHX_ENUMBER);
+          return NULL;
+        }
         if( node_n > max_node_n )
             max_node_n = node_n;
     }
-    if (asprintf(&step_name,"%d",max_node_n+1) >= 0)
+    if (asprintf(&step_name,"%lu",max_node_n+1) >= 0)
         return step_name;
     else
         return NULL;
```